processor/src/main/java/org/mapstruct/ap/internal/util/Executables.java

```diff
@@ -111,9 +110,21 @@ public static boolean isPresenceCheckMethod(Accessor method) {
     public static boolean isSetterMethod(Accessor method) {
         ExecutableElement executable = method.getExecutable();
         return executable != null
-            && isPublic( method )
-            && executable.getParameters().size() == 1
-            && ACCESSOR_NAMING_STRATEGY.getMethodType( executable ) == MethodType.SETTER;
+                && isPublic( method )
+                && executable.getParameters().size() == 1
+                && (ACCESSOR_NAMING_STRATEGY.getMethodType( executable ) == MethodType.SETTER ||
+                isBuilderSetterMethod( method ) );
+    }
+
+    private static boolean isBuilderSetterMethod(Accessor method) {
+        if ( method.isBuilder() ) {
+            String methodReturnType = method.getExecutable().getReturnType().toString();
+            String builderType = method.getBuilderType().toString();
+            return methodReturnType.equals( builderType );
```

Comment:
I'd like to add a bunch of test cases for this feature... I suppose there might be limitations to what features in mapstruct work when using immutable builders?  What are some other things you think I should test that might expose where this approach is going to break?